File: codewave_py/detector.py

class Detector():
	def __init__(self,data={}):
		self.data = data
	def detect(self,finder):
		pass
		

class LangDetector(Detector):
	def detect(self,finder):
		if finder.codewave is not None :
			lang = finder.codewave.editor.getLang()
			if lang is not None :
				return lang.lower()

File: lib/detectors/Detector.js

var Detector = class Detector {
  constructor(data = {}) {
    this.data = data;
  }

  detect(finder) {
    if (this.detected(finder)) {
      if (this.data.result != null) {
        return this.data.result;
      }
    } else {
      if (this.data.else != null) {
        return this.data.else;
      }
    }
  }

  detected(finder) {}

};
exports.Detector = Detector;

